nomad/heartbeat.go

```diff
@@ -181,8 +180,8 @@ func (h *nodeHeartbeater) disconnectState(id string) (bool, bool) {
 		return false, false
 	}
 
-	// Exit if this is the node already in a state other than ready.
-	if node.Status != structs.NodeStatusReady {
+	// Exit if the node is already down.
+	if node.Status == structs.NodeStatusDown {
 		return false, false
 	}
```

Comment:
Mini optimization opportunity here. There are (now) 4 node states, and the original intent here included `NodeStatusInitializing`. Can we still return early from that so we don't have to do a state store lookup?

Reply:
Yes!